torchaudio/prototype/ctc_decoder.py

```diff
@@ -0,0 +1,222 @@
+import torch
```

Comment:
The factory function is not imported in ``prototype/__init__.py``.  Could you add ``kenlm_lexicon_decoder`` to it?

Reply:
`prototype/__init__.py` should not import anything (with external library dependencies).
I missed it when I was reviewing emformer/rnn-t, but they are pure-python implementation, and they won't cause segmentation fault, so they are exceptionally okay.

If ctcdecoder is imported in `prototype/__init__.py`, then KenLM becomes mandatory dependency for using anything in `prototype` module. It will force user who do not care about this decoder but want to use other prototype module to install KenLM, which is inconvenient.

Reply:
I see. If users want to try prototype feature, they need to add it to ``__init__.py`` manually?

Reply:
Nope. If we put everything in a separate submodule under `prototype`, and as long as `prototype.__init__.py` does not import this submodule, users should be able to perform import normally (`from torchaudio.prototype.ctc_decoder import foo`), while we mitigate the risk of broken dependency.

```
torchaudio
└── prototype
    ├── __init__.py  # does not import ctc_decoder module
    └── ctc_decoder
        ├── __init__.py  # Initialize extension module here
        └── ctc_decoder.py
```

Reply:
Got it. Sounds good to me.